docs/source/conf.py

```diff
@@ -50,6 +50,7 @@
 # ones.
 extensions = [
     "sphinx.ext.autodoc",
+    "sphinx_autodoc_typehints",
```

Comment:
This is the thing that allows us to not type hint parameters and returns in the docstring even though we already have them typed in the signature.